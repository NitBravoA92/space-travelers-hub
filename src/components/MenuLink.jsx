import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const MenuLink = ({ url, pageName }) => {
  const navLinkStatus = ({ isActive, isPending }) => {
    if (isActive) return 'active';
    if (isPending) return 'pending';
    return null;
  };

  return (
    <li className="nav-item">
      <NavLink
        to={url}
        className={navLinkStatus}
      >
        {pageName}
      </NavLink>
    </li>
  );
};

MenuLink.propTypes = {
  pageName: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default MenuLink;
