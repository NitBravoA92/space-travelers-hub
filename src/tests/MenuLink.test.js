import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import MenuLink from '../components/MenuLink';

describe('The MenuLink component', () => {
  test("renders a link element (<a>) into the DOM with the title 'Rockets' ", () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <MenuLink url="/rockets" pageName="Rockets" />
        </MemoryRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  describe('should render a link element (<a>) with', () => {
    test("the title 'Missions' and a link pointing to the Missions page", () => {
      const componentLink = render(
        <MemoryRouter>
          <MenuLink url="/missions" pageName="Missions" />
        </MemoryRouter>,
      );

      const linkElement = componentLink.container.querySelector('a');

      expect(componentLink.container).toHaveTextContent('Missions');
      expect(linkElement).toHaveAttribute('href', '/missions');
    });

    test("the title 'My Profile' and a link pointing to the My Profile page", () => {
      const componentLink = render(
        <MemoryRouter>
          <MenuLink url="/my-profile" pageName="My Profile" />
        </MemoryRouter>,
      );

      const link = componentLink.container.querySelector('a');

      expect(componentLink.container).toHaveTextContent('My Profile');
      expect(link).toHaveAttribute('href', '/my-profile');
    });
  });
});
