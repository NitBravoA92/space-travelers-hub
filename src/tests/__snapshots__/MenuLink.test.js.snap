// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MenuLink component renders a link element (<a>) into the DOM with the title 'Rockets'  1`] = `
<li
  className="nav-item"
>
  <a
    className={null}
    href="/rockets"
    onClick={[Function]}
  >
    Rockets
  </a>
</li>
`;
